packages/styled-components/src/utils/hyphenateStyleName.js

```diff
@@ -5,8 +5,10 @@
  * https://github.com/facebook/fbjs/blob/master/packages/fbjs/src/core/hyphenateStyleName.js
  */
 
-const uppercasePattern = /([A-Z])/g;
+const uppercaseCheck = /([A-Z])/;
+const uppercasePattern = new RegExp(uppercaseCheck, 'g');
```

Comment:
Just a quick note here for a minor change; with gzipping it's likely that this will end up being smaller if we just repeat the pattern (with the added `g` flag) but I realise that this was added in an attempt of deduplication.
I can also confirm that this can have a rather unexpected speed up, due to the reduced internal state, matching state, and `lastIndex` travelling over from the native irregexp engine.

Reply:
Yeah, my original attempt also duplicated the regex, so if that's preferable I can put together a PR.

I have no idea if JS engines treat `/foo/g` equivalently to `new RegExp(/foo/, 'g')` or not - I appreciate that regex performance can be sensitive to slight changes.